nixos/modules/services/x11/display-managers/default.nix

```diff
@@ -109,9 +109,8 @@ let
       ${config.systemd.package}/bin/systemctl --user start graphical-session.target
 
       # Allow the user to setup a custom session type.
-      if test -x ~/.xsession; then
-          exec ~/.xsession
-      fi
+      test -x ~/.nix-profile/bin/xsession && exec ~/.nix-profile/bin/xsession
+      test -x ~/.xsession && exec ~/.xsession
```

Comment:
I'd rather have the style stay the same, so to keep the `if`, as it's done in the rest of the file